tls/s2n_config.c

```diff
@@ -683,6 +722,127 @@ int s2n_config_get_cert_type(struct s2n_config *config, s2n_cert_type *cert_type
     notnull_check(config->cert_and_key_pairs->cert_chain.head);
 
     *cert_type = config->cert_and_key_pairs->cert_chain.head->cert_type;
-    
+
+    return 0;
+}
+
+int s2n_config_set_session_state_lifetime(struct s2n_config *config,
+                                          uint64_t lifetime_in_secs)
+{
+    notnull_check(config);
+
+    config->session_state_lifetime_in_nanos = (lifetime_in_secs * ONE_SEC_IN_NANOS);
+    return 0;
+}
+
+int s2n_config_set_session_tickets_onoff(struct s2n_config *config, uint8_t enabled)
+{
+    notnull_check(config);
+
+    if ((config->client_cert_auth_type != S2N_CERT_AUTH_NONE) && enabled) {
+        /* s2n does not support handshakes with CLIENT_AUTH when in session resumption mode */
+        S2N_ERROR(S2N_ERR_CLIENT_AUTH_NOT_SUPPORTED_IN_SESSION_RESUMPTION_MODE);
+    }
+
+    config->use_tickets = enabled;
```

Comment:
Actually, should we just do nothing below if use_tickets == enabled before this one

Reply:
Oh yeah. Will add that check.